include/zephyr/bluetooth/gap.h

```diff
@@ -832,6 +832,70 @@ enum {
  */
 #define BT_GAP_PER_ADV_INTERVAL_TO_MS(interval) ((interval) * 5 / 4)
 
+/**
+ * @brief Convert milliseconds to advertising interval units (0.625 ms)
+ *
+ * @note If @p _int is not a multiple of the unit, it will round down to nearest. For example
+ * BT_GAP_MS_TO_ADV_INT(23) will become 22.5 milliseconds
+ */
+#define BT_GAP_MS_TO_ADV_INT(_int) ((uint16_t)((_int) / 0.625))
```

Comment:
Is there an advantage of using `5 / 4` and `5 / 8` instead of `1.25` and `0.625`?

Reply:
I would be nice to avoid floats if we can. Some devices dont have an FPU so float math can be expensive.

Reply:
But doesn't `5 / 4` also end up as a float? I don't mind changing it, but `0.625` is IMO more readable than `5 / 8`, but if the fraction is better to use, we should use that.

Reply:
> But doesn't `5 / 4` also end up as a float?

No, it does integer division. [Godbolt](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:___c,selection:(endColumn:1,endLineNumber:12,positionColumn:1,positionLineNumber:12,selectionStartColumn:1,selectionStartLineNumber:12,startColumn:1,startLineNumber:12),source:'%23include+%3Cstdint.h%3E%0A%0Auint16_t+convert(uint16_t+ms)%0A%7B%0A++++return+ms+*+5+/+4%3B%0A%7D%0A%0Auint16_t+convert_float(uint16_t+ms)%0A%7B%0A++++return+ms+*+1.25%3B%0A%7D%0A'),l:'5',n:'1',o:'C+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:carmg1230,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:___c,libs:!(),options:'-march%3Darmv8-m.main++-mfloat-abi%3Dsoft+-Os',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+ARM+GCC+12.3.0+(Editor+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

Reply:
Right, `((interval) * 5 / 4)` works before `interval is multiplied by 5 before the division. 

So the `/ 1.25` I have here should become `* 4 / 5`
and the `/ 0.625` should become `* 8 / 5`

Reply:
Thanks :)